<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <arg name="robot_name"/>

    <!-- Set dynamic_reconfigure parameters -->
    <include file="$(find ghost_bridge)/launch/dynamic_reconfigure_params.launch" >
        <arg name="robot_name" value="$(arg robot_name)"/>
    </include>

    <arg name="perception_fps" default="1"/>

    <!-- Perception -->
    <include file="$(find ros_people_model)/launch/perception.launch">
        <arg name="perception_fps" value="$(arg perception_fps)"/>
    </include>

    <!-- Models -->
    <include file="$(find ros_people_model)/launch/models.launch">
        <arg name="perception_fps" value="$(arg perception_fps)"/>
    </include>

    <!-- Face tracker -->
    <node name="face_tracker" pkg="ghost_bridge" type="face_tracker_node.py" output="screen"/>

     <!-- Face tracker -->
    <node name="mirroring" pkg="ros_people_model" type="mirroring.py" output="screen"/>

    <!-- Ghost Bridge -->
    <node name="faces_throttler" type="throttle" pkg="topic_tools" args="messages /faces 2 /faces_throttled" output="screen"/>
    <node name="ghost_bridge" pkg="ghost_bridge" type="ghost_bridge_node.py" output="screen"/>
</launch>
